streams/src/main/java/org/apache/kafka/streams/state/internals/SegmentIterator.java

```diff
@@ -74,7 +74,7 @@ public boolean hasNext() {
             close();
             currentSegment = segments.next();
             try {
-                if (from == null || to == null) {
+                if (from == null && to == null) {
                     if (forward) {
                         currentIterator = currentSegment.all();
```

Comment:
side fix: the `all` case should be  `from == null && to == null`. Otherwise, call `range` method, which is already supported null range query in KIP-763

Reply:
Isn't it sufficient to distinguish the forward and reverse cases and just call range(from, to) or reverseRange(from, to)?

Reply:
Good suggestion. Updated